### ISSUE TITLE
Youtube won't play some videos

### ISSUE THREAD

**Issue by colejd** [ccrama/Slide-iOS#468]
This one is interesting. Some videos just won't play at all, and the youtube player will say the video is broken or can't be found. That can be fixed by passing in the following params:

```
"enablejsapi": 1,
"origin": "http://www.youtube.com",
```

I think this may be against the terms of service though. We should look into this.

**Comment by ccrama** [comment]
If the video is set to not play on external domains or embedded, this will happen. I don't think there is a way around this

**Comment by colejd** [comment]
I've encountered a video that this fixed. I imagine there are others. For those videos that have embedding disabled, we should figure out how to handle it -- surely there's a way.

**Comment by ccrama** [comment]
It's usually done that way for copyright reasons. It's best to not try to get around it, the user can still click in to get to the YouTube app if they want to watch it!

**Comment by colejd** [comment]
Oh yeah, I mean presenting a state to tell the user that the video has embedding disabled or just automatically redirecting to the youtube app / browser.

**Comment by ccrama** [comment]
You mentioned in the OP "the youtube player will say the video is broken or can't be found", is that enough for the user to know to take action?

**Comment by colejd** [comment]
I don't think so. If it's broken I assume the video itself has a problem, not our video player. It would be good for us to detect if that's actually the case.

**Comment by ccrama** [comment]
Do you have an example link that does not work currently?

**Comment by colejd** [comment]
https://reddit.com/r/Music/comments/9edvfn/fatboy_slim_weapon_of_choice_pophiphop/

**Comment by colejd** [comment]
It says "Video unavailable". Which isn't our fault, but I wonder if we can figure out how to tell the user that the video can still be viewed in-app, or better yet automatically redirect them.
